pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -97,7 +96,7 @@ func NewEndpointController(podInformer coreinformers.PodInformer, serviceInforme
 		UpdateFunc: func(old, cur interface{}) {
 			e.enqueueService(cur)
 		},
-		DeleteFunc: e.enqueueService,
+		DeleteFunc: e.dequeueService,
```

Comment:
I added comments on the EndpointSlice controller that also apply throughout this controller.